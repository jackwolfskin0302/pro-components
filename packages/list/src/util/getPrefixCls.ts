const getPrefixCls = (suffixCls: string, customizePrefixCls?: string) => {
  const prefixCls = 'ant-pro';

  if (customizePrefixCls) return customizePrefixCls;

  return suffixCls ? `${prefixCls}-${suffixCls}` : prefixCls;
};

export default getPrefixCls;
