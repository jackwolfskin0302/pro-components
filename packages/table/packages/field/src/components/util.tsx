/**
 * 检查值是否存在
 * 为了 避开 0 和 false
 * @param value
 */
export const checkUndefinedOrNull = (value: any) =>
  value !== undefined && value !== null;

/**
 * 删除对象中所有的空值
 * @param obj
 */
export const removeObjectNull = (obj: { [key: string]: any }) => {
  const newObj = {};
  Object.keys(obj).forEach(key => {
    if (obj[key]) {
      newObj[key] = obj[key];
    }
  });
  return newObj;
};
