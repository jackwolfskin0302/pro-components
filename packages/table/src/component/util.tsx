/* eslint-disable @typescript-eslint/naming-convention */
import React, { ReactText } from 'react';
import { DataIndex } from 'rc-table/lib/interface';

/**
 * 检查值是否存在
 * 为了 避开 0 和 false
 * @param value
 */
export const checkUndefinedOrNull = (value: any) => value !== undefined && value !== null;

/**
 *  根据 key 和 dataIndex 生成唯一 id
 * @param key 用户设置的 key
 * @param dataIndex 在对象中的数据
 * @param index 序列号，理论上唯一
 */
export const genColumnKey = (
  key?: React.ReactText | undefined,
  dataIndex?: DataIndex,
  index?: number,
) => {
  if (key) {
    return key;
  }
  if (!key && dataIndex) {
    if (Array.isArray(dataIndex)) {
      return dataIndex.join('-');
    }
    return dataIndex;
  }
  return `${index}`;
};

/**
 * 删除对象中所有的空值
 * @param obj
 */
export const removeObjectNull = (obj: { [key: string]: any }) => {
  const newObj = {};
  Object.keys(obj).forEach((key) => {
    if (obj[key]) {
      newObj[key] = obj[key];
    }
  });
  return newObj;
};

/**
 * 减少 width，支持 string 和 number
 */
export const reduceWidth = (width?: string | number): string | number | undefined => {
  if (width === undefined) {
    return width;
  }
  if (typeof width === 'string') {
    if (!width.includes('calc')) {
      return `calc(100% - ${width})`;
    }
    return width;
  }
  if (typeof width === 'number') {
    return (width as number) - 32;
  }
  return width;
};
