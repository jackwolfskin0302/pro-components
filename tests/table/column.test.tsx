import { mount } from 'enzyme';
import React from 'react';
import ProTable from '@ant-design/pro-table';
import { request } from './demo';
import { waitForComponentToPaint } from '../util';

describe('Table ColumnSetting', () => {
  it('🎏 render', async () => {
    const callBack = jest.fn();
    const html = mount(
      <ProTable
        size="small"
        columns={[
          {
            title: 'Name',
            key: 'name',
            dataIndex: 'name',
            render: (text) => callBack(text),
          },
        ]}
        request={request}
        rowKey="key"
      />,
    );
    await waitForComponentToPaint(html, 200);
    expect(callBack).toBeCalled();
    expect(callBack).toBeCalledWith('Edward King 0');
  });

  it('🎏 render text', async () => {
    const callBack = jest.fn();
    const html = mount(
      <ProTable
        size="small"
        columns={[
          {
            title: 'Name',
            key: 'name',
            dataIndex: 'name',
            renderText: (text) => callBack(text),
          },
        ]}
        request={request}
        rowKey="key"
      />,
    );
    await waitForComponentToPaint(html, 200);
    expect(callBack).toBeCalled();
    expect(callBack).toBeCalledWith('Edward King 0');
  });

  it('🎏 change text by renderText', async () => {
    const html = mount(
      <ProTable
        size="small"
        columns={[
          {
            title: 'Name',
            key: 'name',
            dataIndex: 'name',
            renderText: (text) => `${text}2144`,
          },
        ]}
        search={false}
        request={request}
        rowKey="key"
      />,
    );
    await waitForComponentToPaint(html, 200);
    expect(html.find('td.ant-table-cell')).toMatchSnapshot();
  });
});
