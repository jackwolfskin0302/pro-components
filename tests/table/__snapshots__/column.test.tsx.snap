// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table ColumnSetting üéè change text by renderText 1`] = `
Array [
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 02144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 12144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 22144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 32144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 42144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 52144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 62144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 72144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 82144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 92144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 102144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 112144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 122144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 132144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 142144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 152144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 162144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 172144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 182144
  </td>,
  <td
    className="ant-table-cell"
    colSpan={null}
    rowSpan={null}
    style={Object {}}
  >
    Edward King 192144
  </td>,
]
`;
